include/fmt/format-inl.h

```diff
@@ -148,7 +147,7 @@ FMT_FUNC void format_error_code(detail::buffer<char>& out, int error_code,
   if (message.size() <= inline_buffer_size - error_code_size)
     format_to(it, FMT_STRING("{}{}"), message, SEP);
   format_to(it, FMT_STRING("{}{}"), ERROR_STR, error_code);
-  assert(out.size() <= inline_buffer_size);
+  FMT_ASSERT(out.size() <= inline_buffer_size, "must not require dynamic alloc");
```

Comment:
I don't think we need assertion messages in any of these cases, I suggest keeping them empty (we should really make the message optional in `FMT_ASSERT` but not in this PR).